### ISSUE TITLE
Keras Not Learning with 3-Channel Images

### ISSUE THREAD

**Issue by thekatiebr** [keras-team/keras#14015]
**System information**  
- Have I written custom code (as opposed to using example directory):  Yes. Files are attached.
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  1.15.0
- Keras version:  2.2.4
- Python version:  3.6.10
- CUDA/cuDNN version:  CUDA: 10.0/ cuDNN: 7.6.4.38
- GPU model and memory:  NVIDA GeForce RTX 2070 EVGA 8GB

**Context**
I’ve been doing image classification for a few years now, but I am by no means an expert. I’m working with 3 datasets: MNIST, [Dogs vs. Cats](https://www.kaggle.com/c/dogs-vs-cats) , and [Diabetic Retinopathy detection](https://www.kaggle.com/c/diabetic-retinopathy-detection), 
converted to binary classification. I definitely hope this is an easy fix!

**Describe the current behavior**  
When I train the neural network on a dataset with three channels (naturally, or converted from grayscale), it doesn’t train correctly, and produces two distinct behaviors depending on the value of the learning rate. If the learning rate is normal ( lr > 0.0001), training accuracy and validation accuracy hover around 50%, and training and validation loss hover around some value (and does not improve). When evaluated on the test set, the network guesses all 1s (same predicted probability greater than 0.5).  If the learning rate is quite low (lr <= 0.0001), the neural network appears to train based on training and validation metrics; however, when evaluated on the test set, the network randomly guesses, and the accuracy is about 50%. 

This happens on all three datasets when represented as 3-channel RGB images. I’m currently using a Keras Image Data Generator, but this also occurs manually converting the images into a NumPy X,y dataset to use Keras model.fit(). I was able to resolve the issue for MNIST by using 1 channel to represent the image, and train the model successfully, but I hope to avoid that for the other two datasets.

This occurs independently of hardware and OS as well, since I have also tested my code on a CentOS 7.7 machine that uses an NVIDIA K80 GPU. 

**Describe the expected behavior**  
What I expect to happen (particularly with the cats vs dogs and MNIST datasets) is the model to learn from the training data and be able to generalize to the test data reasonably well (90% accuracy). Similarly for the diabetic retinopathy data, but I’m okay with 80% accuracy. 

**Code to reproduce the issue**  
I've attached both the neural network code and data generator code. Create a data directory, and a subdirectory entitled "dogs-vs-cats" for dogs vs cats data or "DIABETIC_RETINOPATHY" to store the diabetic retinopathy data. The images are stored in a further subdirectory "train_kaggle2". The CSV dictionaries are in the same directory as "train_kaggle2". My source code is zipped.
[code.zip](https://github.com/keras-team/keras/files/4542339/code.zip)

Execute `python VGG19_DR_Keras.py` to execute the code 

**Other info / logs**  
I've attached the logs with learning rate of 0.01 and 0.0001 in 001.txt and 00001.txt, respectively.
[out_00001.txt](https://github.com/keras-team/keras/files/4542327/out_00001.txt)
[out_001.txt](https://github.com/keras-team/keras/files/4542351/out_001.txt)

**Comment by kumariko** [comment]
@thekatiebr We are checking to see if you still need help on this issue, as you are using an older version of tensorflow(1.x) which is officially considered as end of life. We recommend that you upgrade to 2.6 which is latest stable version of TF and let us know if the issue still persists in newer versions.Thanks!
